src/app/tests/integration/chip_im_initiator.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a chip-im-initiator, for the
+ *      CHIP Interaction Data Model Protocol.
+ *
+ *      Currently it provides simple command sender with sample cluster and command
+ *
+ */
+
+#include "common.h"
+
+#include <app/CommandHandler.h>
+#include <app/CommandSender.h>
+#include <app/InteractionModelEngine.h>
+#include <core/CHIPCore.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#include <support/ErrorStr.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/SecurePairingSession.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/raw/UDP.h>
+
+#define IM_CLIENT_PORT (CHIP_PORT + 1)
+
+namespace chip {
+namespace app {
+InteractionModelEngine sInteractionModelEngine;
+InteractionModelEngine * InteractionModelEngine::GetInstance()
```

Comment:
Why is this test defining a function declared inside of CHIP? That implies there's a missing or circular dependency.

Reply:
agree, just relocate to InteractionModelEngine.cpp